lib/bind.js

```diff
@@ -19,7 +19,16 @@ function bind(fn, context, args) {
     assert.optionalArray(args, 'args');
 
     return function boundFn() {
-        fn.apply(context || null, args || []);
+
+        // working with arguments is a bit hazardous in terms of deopt.
+        // https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments
```

Comment:
Fascinating, the things you learn! 

Honestly it's probably a wash, but for my own curiosity I'd love to do a quick benchmark between 0.12/4/5 just to see if it's even worth the trouble. Otherwise, happy to go with either the classic slice approach or the one you've suggested here.